src/amo/components/SponsoredAddonsShelf/index.js

```diff
@@ -72,23 +102,44 @@ export class SponsoredAddonsShelfBase extends React.Component<InternalProps> {
 
   render() {
     const {
+      _config,
       addonInstallSource,
       className,
+      errorHandler,
       i18n,
+      isLoading,
       resultsLoaded,
       shelves,
+      shelfData,
     } = this.props;
 
-    const { promotedExtensions } = shelves;
-    if (Array.isArray(promotedExtensions) && !promotedExtensions.length) {
+    const useAdzerk = _config.get('enableFeatureUseAdzerkForSponsoredShelf');
+    if (useAdzerk && errorHandler.hasError()) {
+      log.debug(
+        'Error when fetching sponsored add-ons, hiding the SponsoredAddonsShelf component.',
+      );
+      return null;
+    }
+
+    let sponsoredAddons;
+
+    if (useAdzerk) {
+      if (shelfData) {
+        sponsoredAddons = shelfData.addons;
+      }
+    } else if (shelves) {
+      sponsoredAddons = shelves.promotedExtensions;
+    }
```

Comment:
This could maybe move to its own method like `getSponsoredAddons()`

Reply:
It could, but the code is only ever used once, so what does that buy us? All we'd really be doing is moving code from one place to another, and adding a function. Plus, we can get rid of this once we turn on the Adzerk integration everywhere (and no longer need the flag). Even if we move away from Adzerk, we'd still want to use the new `promoted` endpoint, so I think this code is definitely temporary.

Reply:
Readability

Reply:
I'm not sure how much more readable it would be to simply take the code and more it into a function. We'll still have the nested ifs. I'm inclined to just leave it as is, and then remove it later when we get rid of the flag. Are you okay with that?

Reply:
Yep, that was a suggestion but it's a nit